packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -334,6 +336,7 @@ export default class NewVm extends BaseComponent {
       cpuWeight: '',
       existingDisks: {},
       fastClone: true,
+      hvmBootFirmware: '',
```

Comment:
Empty string is not a valid boot firmware value.

Reply:
I'll put `bios` as a default value.

Reply:
I'm not sure it's related to your review, but Citrix doc said:

> If you do not specify the mode, it defaults to ‘bios’.

Reply:
@olivierlambert Thank you for the information. Not setting the VM `firmware` field if the user not select `uefi` is also an option to consider.
My review was about not setting the `firmware` field to an empty string because is not supported by `CH/XCP-ng `.

Reply:
Maybe we should set either `undefined` or `'uefi`' in `VM.HVM_boot_params.firmware` then?

Reply:
I agree with you :+1:  @olivierlambert  i don't know if the boot default value can change per version. But, to avoid this case, i suggest to display to the users `default(bios)` instead of `bios` to say that generally the default value is `bios`.

Reply:
Maybe there should be 3 values, *default (BIOS)* (`undefined`), *BIOS* (`bios`) and *UEFI* (`uefi`).

This will allow full control by the users and will behave correctly if one day the default value changed in the XAPI.

Reply:
Waiting for @olivierlambert to confirm this :)

Reply:
Why not 3 values, but it's more complex for the user: what is "empty"? why? IDK if there is a good way to expose this without confusing people.

Reply:
The user will not see *empty*, this will be the value in the XAPI, the user will see *default (BIOS)*.

Reply:
Okay so LGTM then

Reply:
I'm agree with @olivierlambert. There's two values `bios` (default value) and `uefi` ... 
I don't think that the default value in the XAPI will change in short term

Reply:
Let's go with three options for now, we'll change it if necessary based on user feedback.